src/Support/TypeToSchemaExtensions/ResponderTypeToSchema.php

```diff
@@ -0,0 +1,230 @@
+                    && (new TypeWalker)->first($item->value, fn (Type $t) => $t->isInstanceOf(MissingValue::class))
+                ) {
+                    $newType = array_filter($item->value->types, fn (Type $t) => !$t->isInstanceOf(MissingValue::class));
+
+                    if (!count($newType)) {
+                        return [];
+                    }
+
+                    $item->isOptional = true;
+
+                    if (count($newType) === 1) {
+                        $item->value = $newType[0];
+
+                        return $this->flattenMergeValues([$item]);
+                    }
+
+                    $item->value = new Union($newType);
+
+                    return $this->flattenMergeValues([$item]);
+                }
+
+                if (
+                    $item->value instanceof Generic
+                    && $item->value->isInstanceOf(MergeValue::class)
+                ) {
+                    $arrayToMerge = $item->value->templateTypes[1];
+
+                    // Second generic argument of the `MergeValue` class must be an array.
+                    // Otherwise, we ignore it from the resulting array.
+                    if (!$arrayToMerge instanceof ArrayType) {
+                        return [];
+                    }
+
+                    $arrayToMergeItems = $this->flattenMergeValues($arrayToMerge->items);
+
+                    $mergingArrayValuesShouldBeRequired = $item->value->templateTypes[0] instanceof LiteralBooleanType
+                        && $item->value->templateTypes[0]->value === true;
+
+                    if (!$mergingArrayValuesShouldBeRequired || $item->isOptional) {
+                        foreach ($arrayToMergeItems as $mergingItem) {
+                            $mergingItem->isOptional = true;
+                        }
+                    }
+
+                    return $arrayToMergeItems;
+                }
+
+                return [$item];
+            })
+            ->values()
+            ->all();
+    }
+
+    /**
+     * @param  Generic  $type
+     */
+    public function toResponse(Type $type)
+    {
+        $definition = $this->infer->analyzeClass($type->name);
+
+        $additional = $type->templateTypes[1 /* TAdditional */] ?? new UnknownType();
+
+        $openApiType = $this->openApiTransformer->transform($type);
+
+        if (($withArray = $definition->getMethodCallType('with')) instanceof ArrayType) {
+            $withArray->items = $this->flattenMergeValues($withArray->items);
+        }
+        if ($additional instanceof ArrayType) {
+            $additional->items = $this->flattenMergeValues($additional->items);
+        }
+
+        $shouldWrap = ($wrapKey = $type->name::$wrap ?? null) !== null
+            || $withArray instanceof ArrayType
+            || $additional instanceof ArrayType;
+        $wrapKey = $wrapKey ?: 'data';
+
+        if ($shouldWrap) {
+            $openApiType = (new \Dedoc\Scramble\Support\Generator\Types\ObjectType())
```

Comment:
```suggestion
            $openApiType = (new \Dedoc\Scramble\Support\Generator\Types\ObjectType)
```
You can skip the `()` when you don't pass anything to the constructor.